src/validators.test.ts

```diff
@@ -66,3 +68,102 @@ describe('length', () => {
     expect(got).toEqual(expectedError);
   });
 });
+
+// Pattern validator
+describe('pattern', () => {
+  it('returns error', () => {
+    const expectedError = `Invalid characters in input`;
```

Comment:
The error msg could have been better.
E.g. `The value 'blah' does not match the pattern 'blah'